projects/angular/src/utils/animations/expandable-animation/base-expandable-animation.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2016-2025 Broadcom. All Rights Reserved.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { ElementRef, Renderer2 } from '@angular/core';
+
+import { DomAdapter } from '../../dom-adapter/dom-adapter';
+
+export class BaseExpandableAnimation {
+  startHeight = 0;
+
+  constructor(
+    protected element: ElementRef<HTMLElement>,
+    protected domAdapter: DomAdapter,
+    protected renderer: Renderer2
+  ) {}
+
+  updateStartHeight() {
+    this.startHeight = this.domAdapter.computedHeight(this.element.nativeElement) || 0;
+  }
+
+  initAnimationEffects() {
+    this.renderer.setStyle(this.element.nativeElement, 'overflow', 'hidden');
+  }
+
+  cleanupAnimationEffects(cancelAnimations = false) {
+    this.renderer.removeStyle(this.element.nativeElement, 'overflow');
+
+    // A "safe" auto-update of the height ensuring basic OOTB user experience .
+    // Prone to small jumps in initial animation height if data was changed in the meantime, window was resized, etc.
```

Comment:
Move from my other comment for the code comment.

```suggestion
    // A "safe" auto-update of the height ensuring basic OOTB user experience.
    // Prone to small jumps in initial animation height if data was changed in the meantime, the window was resized, etc.
```

Reply:
Fixed in [0a62d89](https://github.com/vmware-clarity/ng-clarity/pull/1827/commits/0a62d891696a702894a228b740b0d0a5290db7e5)